src/app/js/formats/streamgraph/utils.js

```diff
@@ -0,0 +1,159 @@
+import differenceBy from "lodash/differenceBy";
+import * as d3 from "d3";
+
+export function zoomFunction() {
+  // create new scale ojects based on event
+  let new_xScale = d3.event.transform.rescaleX(this.xAxisScale);
```

Comment:
Instead of binding `this` to this function, you have two solutions:
- Pass all what you needs in argument
- Declare this function as a method of the component class

I'd prefer that you use the second option.

Reply:
I don't understand why but when I move the declaration of this function in the class through a method it fails, the method have some problems with this.xAxisScale which is empty. I lost a lot of time on it so I will let the function in this state for now.